hudi-common/src/main/java/org/apache/hudi/common/util/CleanerUtils.java

```diff
@@ -107,21 +109,18 @@ public static HoodieCleanMetadata convertCleanMetadata(String startCleanTime,
    */
   public static HoodieCleanMetadata getCleanerMetadata(HoodieTableMetaClient metaClient, HoodieInstant cleanInstant)
       throws IOException {
-    CleanMetadataMigrator metadataMigrator = new CleanMetadataMigrator(metaClient);
-    HoodieCleanMetadata cleanMetadata = TimelineMetadataUtils.deserializeHoodieCleanMetadata(
-        metaClient.getActiveTimeline().readCleanerInfoAsBytes(cleanInstant).get());
-    return metadataMigrator.upgradeToLatest(cleanMetadata, cleanMetadata.getVersion());
+    HoodieCleanMetadata cleanMetadata = metaClient.getActiveTimeline().loadHoodieCleanMetadata(cleanInstant);
+    return upgradeCleanMetadata(metaClient, cleanMetadata);
```

Comment:
nit: We should merge these cleaner metadata utils to `HoodieTimeline` class.  We can take that up in a follow-up PR with a new JIRA ticket.

Reply:
HUDI-9091 to track